{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "ename": "ConnectionError",
     "evalue": "HTTPSConnectionPool(host='www.soaringspot.com', port=443): Max retries exceeded with url: /en_gb/junior-world-gliding-championships-2022-tabor-2022/results (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x1054b8220>: Failed to establish a new connection: [Errno 8] nodename nor servname provided, or not known'))",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mgaierror\u001b[0m                                  Traceback (most recent call last)",
      "File \u001b[0;32m/opt/homebrew/lib/python3.9/site-packages/urllib3/connection.py:174\u001b[0m, in \u001b[0;36mHTTPConnection._new_conn\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m    173\u001b[0m \u001b[39mtry\u001b[39;00m:\n\u001b[0;32m--> 174\u001b[0m     conn \u001b[39m=\u001b[39m connection\u001b[39m.\u001b[39;49mcreate_connection(\n\u001b[1;32m    175\u001b[0m         (\u001b[39mself\u001b[39;49m\u001b[39m.\u001b[39;49m_dns_host, \u001b[39mself\u001b[39;49m\u001b[39m.\u001b[39;49mport), \u001b[39mself\u001b[39;49m\u001b[39m.\u001b[39;49mtimeout, \u001b[39m*\u001b[39;49m\u001b[39m*\u001b[39;49mextra_kw\n\u001b[1;32m    176\u001b[0m     )\n\u001b[1;32m    178\u001b[0m \u001b[39mexcept\u001b[39;00m SocketTimeout:\n",
      "File \u001b[0;32m/opt/homebrew/lib/python3.9/site-packages/urllib3/util/connection.py:72\u001b[0m, in \u001b[0;36mcreate_connection\u001b[0;34m(address, timeout, source_address, socket_options)\u001b[0m\n\u001b[1;32m     68\u001b[0m     \u001b[39mreturn\u001b[39;00m six\u001b[39m.\u001b[39mraise_from(\n\u001b[1;32m     69\u001b[0m         LocationParseError(\u001b[39mu\u001b[39m\u001b[39m\"\u001b[39m\u001b[39m'\u001b[39m\u001b[39m%s\u001b[39;00m\u001b[39m'\u001b[39m\u001b[39m, label empty or too long\u001b[39m\u001b[39m\"\u001b[39m \u001b[39m%\u001b[39m host), \u001b[39mNone\u001b[39;00m\n\u001b[1;32m     70\u001b[0m     )\n\u001b[0;32m---> 72\u001b[0m \u001b[39mfor\u001b[39;00m res \u001b[39min\u001b[39;00m socket\u001b[39m.\u001b[39;49mgetaddrinfo(host, port, family, socket\u001b[39m.\u001b[39;49mSOCK_STREAM):\n\u001b[1;32m     73\u001b[0m     af, socktype, proto, canonname, sa \u001b[39m=\u001b[39m res\n",
      "File \u001b[0;32m/opt/homebrew/Cellar/python@3.9/3.9.13_1/Frameworks/Python.framework/Versions/3.9/lib/python3.9/socket.py:954\u001b[0m, in \u001b[0;36mgetaddrinfo\u001b[0;34m(host, port, family, type, proto, flags)\u001b[0m\n\u001b[1;32m    953\u001b[0m addrlist \u001b[39m=\u001b[39m []\n\u001b[0;32m--> 954\u001b[0m \u001b[39mfor\u001b[39;00m res \u001b[39min\u001b[39;00m _socket\u001b[39m.\u001b[39;49mgetaddrinfo(host, port, family, \u001b[39mtype\u001b[39;49m, proto, flags):\n\u001b[1;32m    955\u001b[0m     af, socktype, proto, canonname, sa \u001b[39m=\u001b[39m res\n",
      "\u001b[0;31mgaierror\u001b[0m: [Errno 8] nodename nor servname provided, or not known",
      "\nDuring handling of the above exception, another exception occurred:\n",
      "\u001b[0;31mNewConnectionError\u001b[0m                        Traceback (most recent call last)",
      "File \u001b[0;32m/opt/homebrew/lib/python3.9/site-packages/urllib3/connectionpool.py:703\u001b[0m, in \u001b[0;36mHTTPConnectionPool.urlopen\u001b[0;34m(self, method, url, body, headers, retries, redirect, assert_same_host, timeout, pool_timeout, release_conn, chunked, body_pos, **response_kw)\u001b[0m\n\u001b[1;32m    702\u001b[0m \u001b[39m# Make the request on the httplib connection object.\u001b[39;00m\n\u001b[0;32m--> 703\u001b[0m httplib_response \u001b[39m=\u001b[39m \u001b[39mself\u001b[39;49m\u001b[39m.\u001b[39;49m_make_request(\n\u001b[1;32m    704\u001b[0m     conn,\n\u001b[1;32m    705\u001b[0m     method,\n\u001b[1;32m    706\u001b[0m     url,\n\u001b[1;32m    707\u001b[0m     timeout\u001b[39m=\u001b[39;49mtimeout_obj,\n\u001b[1;32m    708\u001b[0m     body\u001b[39m=\u001b[39;49mbody,\n\u001b[1;32m    709\u001b[0m     headers\u001b[39m=\u001b[39;49mheaders,\n\u001b[1;32m    710\u001b[0m     chunked\u001b[39m=\u001b[39;49mchunked,\n\u001b[1;32m    711\u001b[0m )\n\u001b[1;32m    713\u001b[0m \u001b[39m# If we're going to release the connection in ``finally:``, then\u001b[39;00m\n\u001b[1;32m    714\u001b[0m \u001b[39m# the response doesn't need to know about the connection. Otherwise\u001b[39;00m\n\u001b[1;32m    715\u001b[0m \u001b[39m# it will also try to release it and we'll have a double-release\u001b[39;00m\n\u001b[1;32m    716\u001b[0m \u001b[39m# mess.\u001b[39;00m\n",
      "File \u001b[0;32m/opt/homebrew/lib/python3.9/site-packages/urllib3/connectionpool.py:386\u001b[0m, in \u001b[0;36mHTTPConnectionPool._make_request\u001b[0;34m(self, conn, method, url, timeout, chunked, **httplib_request_kw)\u001b[0m\n\u001b[1;32m    385\u001b[0m \u001b[39mtry\u001b[39;00m:\n\u001b[0;32m--> 386\u001b[0m     \u001b[39mself\u001b[39;49m\u001b[39m.\u001b[39;49m_validate_conn(conn)\n\u001b[1;32m    387\u001b[0m \u001b[39mexcept\u001b[39;00m (SocketTimeout, BaseSSLError) \u001b[39mas\u001b[39;00m e:\n\u001b[1;32m    388\u001b[0m     \u001b[39m# Py2 raises this as a BaseSSLError, Py3 raises it as socket timeout.\u001b[39;00m\n",
      "File \u001b[0;32m/opt/homebrew/lib/python3.9/site-packages/urllib3/connectionpool.py:1042\u001b[0m, in \u001b[0;36mHTTPSConnectionPool._validate_conn\u001b[0;34m(self, conn)\u001b[0m\n\u001b[1;32m   1041\u001b[0m \u001b[39mif\u001b[39;00m \u001b[39mnot\u001b[39;00m \u001b[39mgetattr\u001b[39m(conn, \u001b[39m\"\u001b[39m\u001b[39msock\u001b[39m\u001b[39m\"\u001b[39m, \u001b[39mNone\u001b[39;00m):  \u001b[39m# AppEngine might not have  `.sock`\u001b[39;00m\n\u001b[0;32m-> 1042\u001b[0m     conn\u001b[39m.\u001b[39;49mconnect()\n\u001b[1;32m   1044\u001b[0m \u001b[39mif\u001b[39;00m \u001b[39mnot\u001b[39;00m conn\u001b[39m.\u001b[39mis_verified:\n",
      "File \u001b[0;32m/opt/homebrew/lib/python3.9/site-packages/urllib3/connection.py:358\u001b[0m, in \u001b[0;36mHTTPSConnection.connect\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m    356\u001b[0m \u001b[39mdef\u001b[39;00m \u001b[39mconnect\u001b[39m(\u001b[39mself\u001b[39m):\n\u001b[1;32m    357\u001b[0m     \u001b[39m# Add certificate verification\u001b[39;00m\n\u001b[0;32m--> 358\u001b[0m     \u001b[39mself\u001b[39m\u001b[39m.\u001b[39msock \u001b[39m=\u001b[39m conn \u001b[39m=\u001b[39m \u001b[39mself\u001b[39;49m\u001b[39m.\u001b[39;49m_new_conn()\n\u001b[1;32m    359\u001b[0m     hostname \u001b[39m=\u001b[39m \u001b[39mself\u001b[39m\u001b[39m.\u001b[39mhost\n",
      "File \u001b[0;32m/opt/homebrew/lib/python3.9/site-packages/urllib3/connection.py:186\u001b[0m, in \u001b[0;36mHTTPConnection._new_conn\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m    185\u001b[0m \u001b[39mexcept\u001b[39;00m SocketError \u001b[39mas\u001b[39;00m e:\n\u001b[0;32m--> 186\u001b[0m     \u001b[39mraise\u001b[39;00m NewConnectionError(\n\u001b[1;32m    187\u001b[0m         \u001b[39mself\u001b[39m, \u001b[39m\"\u001b[39m\u001b[39mFailed to establish a new connection: \u001b[39m\u001b[39m%s\u001b[39;00m\u001b[39m\"\u001b[39m \u001b[39m%\u001b[39m e\n\u001b[1;32m    188\u001b[0m     )\n\u001b[1;32m    190\u001b[0m \u001b[39mreturn\u001b[39;00m conn\n",
      "\u001b[0;31mNewConnectionError\u001b[0m: <urllib3.connection.HTTPSConnection object at 0x1054b8220>: Failed to establish a new connection: [Errno 8] nodename nor servname provided, or not known",
      "\nDuring handling of the above exception, another exception occurred:\n",
      "\u001b[0;31mMaxRetryError\u001b[0m                             Traceback (most recent call last)",
      "File \u001b[0;32m/opt/homebrew/lib/python3.9/site-packages/requests/adapters.py:489\u001b[0m, in \u001b[0;36mHTTPAdapter.send\u001b[0;34m(self, request, stream, timeout, verify, cert, proxies)\u001b[0m\n\u001b[1;32m    488\u001b[0m \u001b[39mif\u001b[39;00m \u001b[39mnot\u001b[39;00m chunked:\n\u001b[0;32m--> 489\u001b[0m     resp \u001b[39m=\u001b[39m conn\u001b[39m.\u001b[39;49murlopen(\n\u001b[1;32m    490\u001b[0m         method\u001b[39m=\u001b[39;49mrequest\u001b[39m.\u001b[39;49mmethod,\n\u001b[1;32m    491\u001b[0m         url\u001b[39m=\u001b[39;49murl,\n\u001b[1;32m    492\u001b[0m         body\u001b[39m=\u001b[39;49mrequest\u001b[39m.\u001b[39;49mbody,\n\u001b[1;32m    493\u001b[0m         headers\u001b[39m=\u001b[39;49mrequest\u001b[39m.\u001b[39;49mheaders,\n\u001b[1;32m    494\u001b[0m         redirect\u001b[39m=\u001b[39;49m\u001b[39mFalse\u001b[39;49;00m,\n\u001b[1;32m    495\u001b[0m         assert_same_host\u001b[39m=\u001b[39;49m\u001b[39mFalse\u001b[39;49;00m,\n\u001b[1;32m    496\u001b[0m         preload_content\u001b[39m=\u001b[39;49m\u001b[39mFalse\u001b[39;49;00m,\n\u001b[1;32m    497\u001b[0m         decode_content\u001b[39m=\u001b[39;49m\u001b[39mFalse\u001b[39;49;00m,\n\u001b[1;32m    498\u001b[0m         retries\u001b[39m=\u001b[39;49m\u001b[39mself\u001b[39;49m\u001b[39m.\u001b[39;49mmax_retries,\n\u001b[1;32m    499\u001b[0m         timeout\u001b[39m=\u001b[39;49mtimeout,\n\u001b[1;32m    500\u001b[0m     )\n\u001b[1;32m    502\u001b[0m \u001b[39m# Send the request.\u001b[39;00m\n\u001b[1;32m    503\u001b[0m \u001b[39melse\u001b[39;00m:\n",
      "File \u001b[0;32m/opt/homebrew/lib/python3.9/site-packages/urllib3/connectionpool.py:787\u001b[0m, in \u001b[0;36mHTTPConnectionPool.urlopen\u001b[0;34m(self, method, url, body, headers, retries, redirect, assert_same_host, timeout, pool_timeout, release_conn, chunked, body_pos, **response_kw)\u001b[0m\n\u001b[1;32m    785\u001b[0m     e \u001b[39m=\u001b[39m ProtocolError(\u001b[39m\"\u001b[39m\u001b[39mConnection aborted.\u001b[39m\u001b[39m\"\u001b[39m, e)\n\u001b[0;32m--> 787\u001b[0m retries \u001b[39m=\u001b[39m retries\u001b[39m.\u001b[39;49mincrement(\n\u001b[1;32m    788\u001b[0m     method, url, error\u001b[39m=\u001b[39;49me, _pool\u001b[39m=\u001b[39;49m\u001b[39mself\u001b[39;49m, _stacktrace\u001b[39m=\u001b[39;49msys\u001b[39m.\u001b[39;49mexc_info()[\u001b[39m2\u001b[39;49m]\n\u001b[1;32m    789\u001b[0m )\n\u001b[1;32m    790\u001b[0m retries\u001b[39m.\u001b[39msleep()\n",
      "File \u001b[0;32m/opt/homebrew/lib/python3.9/site-packages/urllib3/util/retry.py:592\u001b[0m, in \u001b[0;36mRetry.increment\u001b[0;34m(self, method, url, response, error, _pool, _stacktrace)\u001b[0m\n\u001b[1;32m    591\u001b[0m \u001b[39mif\u001b[39;00m new_retry\u001b[39m.\u001b[39mis_exhausted():\n\u001b[0;32m--> 592\u001b[0m     \u001b[39mraise\u001b[39;00m MaxRetryError(_pool, url, error \u001b[39mor\u001b[39;00m ResponseError(cause))\n\u001b[1;32m    594\u001b[0m log\u001b[39m.\u001b[39mdebug(\u001b[39m\"\u001b[39m\u001b[39mIncremented Retry for (url=\u001b[39m\u001b[39m'\u001b[39m\u001b[39m%s\u001b[39;00m\u001b[39m'\u001b[39m\u001b[39m): \u001b[39m\u001b[39m%r\u001b[39;00m\u001b[39m\"\u001b[39m, url, new_retry)\n",
      "\u001b[0;31mMaxRetryError\u001b[0m: HTTPSConnectionPool(host='www.soaringspot.com', port=443): Max retries exceeded with url: /en_gb/junior-world-gliding-championships-2022-tabor-2022/results (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x1054b8220>: Failed to establish a new connection: [Errno 8] nodename nor servname provided, or not known'))",
      "\nDuring handling of the above exception, another exception occurred:\n",
      "\u001b[0;31mConnectionError\u001b[0m                           Traceback (most recent call last)",
      "\u001b[1;32m/Users/jonpring/github/gliding/task_analyser.ipynb Cell 1\u001b[0m in \u001b[0;36m<cell line: 6>\u001b[0;34m()\u001b[0m\n\u001b[1;32m      <a href='vscode-notebook-cell:/Users/jonpring/github/gliding/task_analyser.ipynb#ch0000000?line=2'>3</a>\u001b[0m competition_url \u001b[39m=\u001b[39m \u001b[39m\"\u001b[39m\u001b[39m/en_gb/junior-world-gliding-championships-2022-tabor-2022\u001b[39m\u001b[39m\"\u001b[39m\n\u001b[1;32m      <a href='vscode-notebook-cell:/Users/jonpring/github/gliding/task_analyser.ipynb#ch0000000?line=4'>5</a>\u001b[0m jwgc22 \u001b[39m=\u001b[39m ss\u001b[39m.\u001b[39mSoaringSpotDL(competition_url)\n\u001b[0;32m----> <a href='vscode-notebook-cell:/Users/jonpring/github/gliding/task_analyser.ipynb#ch0000000?line=5'>6</a>\u001b[0m jwgc22\u001b[39m.\u001b[39;49mdownload_traces()\n",
      "File \u001b[0;32m~/github/gliding/soaring_spot_dl.py:129\u001b[0m, in \u001b[0;36mSoaringSpotDL.download_traces\u001b[0;34m(self, comp_class, task_day, overwrite, select_day)\u001b[0m\n\u001b[1;32m    128\u001b[0m \u001b[39mdef\u001b[39;00m \u001b[39mdownload_traces\u001b[39m(\u001b[39mself\u001b[39m, comp_class\u001b[39m=\u001b[39m\u001b[39m\"\u001b[39m\u001b[39mall\u001b[39m\u001b[39m\"\u001b[39m, task_day\u001b[39m=\u001b[39m\u001b[39m\"\u001b[39m\u001b[39mall\u001b[39m\u001b[39m\"\u001b[39m, overwrite\u001b[39m=\u001b[39m\u001b[39mFalse\u001b[39;00m, select_day\u001b[39m=\u001b[39m\u001b[39mFalse\u001b[39;00m):\n\u001b[0;32m--> 129\u001b[0m     days \u001b[39m=\u001b[39m \u001b[39mself\u001b[39;49m\u001b[39m.\u001b[39;49m__get_days_flown()\n\u001b[1;32m    130\u001b[0m     \u001b[39mif\u001b[39;00m select_day:\n\u001b[1;32m    131\u001b[0m         day \u001b[39m=\u001b[39m \u001b[39mself\u001b[39m\u001b[39m.\u001b[39m__select_day(days)\n",
      "File \u001b[0;32m~/github/gliding/soaring_spot_dl.py:20\u001b[0m, in \u001b[0;36mSoaringSpotDL.__get_days_flown\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m     18\u001b[0m \u001b[39mdef\u001b[39;00m \u001b[39m__get_days_flown\u001b[39m(\u001b[39mself\u001b[39m):\n\u001b[1;32m     19\u001b[0m     \u001b[39m\"\"\"Retrieve competition classes and task days from soaring spot\"\"\"\u001b[39;00m\n\u001b[0;32m---> 20\u001b[0m     page \u001b[39m=\u001b[39m requests\u001b[39m.\u001b[39;49mget(\u001b[39mself\u001b[39;49m\u001b[39m.\u001b[39;49msoaringspot_url \u001b[39m+\u001b[39;49m \u001b[39mself\u001b[39;49m\u001b[39m.\u001b[39;49mcomp_url \u001b[39m+\u001b[39;49m \u001b[39m\"\u001b[39;49m\u001b[39m/results\u001b[39;49m\u001b[39m\"\u001b[39;49m)\n\u001b[1;32m     21\u001b[0m     soup \u001b[39m=\u001b[39m BeautifulSoup(page\u001b[39m.\u001b[39mcontent, \u001b[39m'\u001b[39m\u001b[39mhtml.parser\u001b[39m\u001b[39m'\u001b[39m)\n\u001b[1;32m     22\u001b[0m     days_flown \u001b[39m=\u001b[39m []\n",
      "File \u001b[0;32m/opt/homebrew/lib/python3.9/site-packages/requests/api.py:73\u001b[0m, in \u001b[0;36mget\u001b[0;34m(url, params, **kwargs)\u001b[0m\n\u001b[1;32m     62\u001b[0m \u001b[39mdef\u001b[39;00m \u001b[39mget\u001b[39m(url, params\u001b[39m=\u001b[39m\u001b[39mNone\u001b[39;00m, \u001b[39m*\u001b[39m\u001b[39m*\u001b[39mkwargs):\n\u001b[1;32m     63\u001b[0m     \u001b[39mr\u001b[39m\u001b[39m\"\"\"Sends a GET request.\u001b[39;00m\n\u001b[1;32m     64\u001b[0m \n\u001b[1;32m     65\u001b[0m \u001b[39m    :param url: URL for the new :class:`Request` object.\u001b[39;00m\n\u001b[0;32m   (...)\u001b[0m\n\u001b[1;32m     70\u001b[0m \u001b[39m    :rtype: requests.Response\u001b[39;00m\n\u001b[1;32m     71\u001b[0m \u001b[39m    \"\"\"\u001b[39;00m\n\u001b[0;32m---> 73\u001b[0m     \u001b[39mreturn\u001b[39;00m request(\u001b[39m\"\u001b[39;49m\u001b[39mget\u001b[39;49m\u001b[39m\"\u001b[39;49m, url, params\u001b[39m=\u001b[39;49mparams, \u001b[39m*\u001b[39;49m\u001b[39m*\u001b[39;49mkwargs)\n",
      "File \u001b[0;32m/opt/homebrew/lib/python3.9/site-packages/requests/api.py:59\u001b[0m, in \u001b[0;36mrequest\u001b[0;34m(method, url, **kwargs)\u001b[0m\n\u001b[1;32m     55\u001b[0m \u001b[39m# By using the 'with' statement we are sure the session is closed, thus we\u001b[39;00m\n\u001b[1;32m     56\u001b[0m \u001b[39m# avoid leaving sockets open which can trigger a ResourceWarning in some\u001b[39;00m\n\u001b[1;32m     57\u001b[0m \u001b[39m# cases, and look like a memory leak in others.\u001b[39;00m\n\u001b[1;32m     58\u001b[0m \u001b[39mwith\u001b[39;00m sessions\u001b[39m.\u001b[39mSession() \u001b[39mas\u001b[39;00m session:\n\u001b[0;32m---> 59\u001b[0m     \u001b[39mreturn\u001b[39;00m session\u001b[39m.\u001b[39;49mrequest(method\u001b[39m=\u001b[39;49mmethod, url\u001b[39m=\u001b[39;49murl, \u001b[39m*\u001b[39;49m\u001b[39m*\u001b[39;49mkwargs)\n",
      "File \u001b[0;32m/opt/homebrew/lib/python3.9/site-packages/requests/sessions.py:587\u001b[0m, in \u001b[0;36mSession.request\u001b[0;34m(self, method, url, params, data, headers, cookies, files, auth, timeout, allow_redirects, proxies, hooks, stream, verify, cert, json)\u001b[0m\n\u001b[1;32m    582\u001b[0m send_kwargs \u001b[39m=\u001b[39m {\n\u001b[1;32m    583\u001b[0m     \u001b[39m\"\u001b[39m\u001b[39mtimeout\u001b[39m\u001b[39m\"\u001b[39m: timeout,\n\u001b[1;32m    584\u001b[0m     \u001b[39m\"\u001b[39m\u001b[39mallow_redirects\u001b[39m\u001b[39m\"\u001b[39m: allow_redirects,\n\u001b[1;32m    585\u001b[0m }\n\u001b[1;32m    586\u001b[0m send_kwargs\u001b[39m.\u001b[39mupdate(settings)\n\u001b[0;32m--> 587\u001b[0m resp \u001b[39m=\u001b[39m \u001b[39mself\u001b[39;49m\u001b[39m.\u001b[39;49msend(prep, \u001b[39m*\u001b[39;49m\u001b[39m*\u001b[39;49msend_kwargs)\n\u001b[1;32m    589\u001b[0m \u001b[39mreturn\u001b[39;00m resp\n",
      "File \u001b[0;32m/opt/homebrew/lib/python3.9/site-packages/requests/sessions.py:701\u001b[0m, in \u001b[0;36mSession.send\u001b[0;34m(self, request, **kwargs)\u001b[0m\n\u001b[1;32m    698\u001b[0m start \u001b[39m=\u001b[39m preferred_clock()\n\u001b[1;32m    700\u001b[0m \u001b[39m# Send the request\u001b[39;00m\n\u001b[0;32m--> 701\u001b[0m r \u001b[39m=\u001b[39m adapter\u001b[39m.\u001b[39;49msend(request, \u001b[39m*\u001b[39;49m\u001b[39m*\u001b[39;49mkwargs)\n\u001b[1;32m    703\u001b[0m \u001b[39m# Total elapsed time of the request (approximately)\u001b[39;00m\n\u001b[1;32m    704\u001b[0m elapsed \u001b[39m=\u001b[39m preferred_clock() \u001b[39m-\u001b[39m start\n",
      "File \u001b[0;32m/opt/homebrew/lib/python3.9/site-packages/requests/adapters.py:565\u001b[0m, in \u001b[0;36mHTTPAdapter.send\u001b[0;34m(self, request, stream, timeout, verify, cert, proxies)\u001b[0m\n\u001b[1;32m    561\u001b[0m     \u001b[39mif\u001b[39;00m \u001b[39misinstance\u001b[39m(e\u001b[39m.\u001b[39mreason, _SSLError):\n\u001b[1;32m    562\u001b[0m         \u001b[39m# This branch is for urllib3 v1.22 and later.\u001b[39;00m\n\u001b[1;32m    563\u001b[0m         \u001b[39mraise\u001b[39;00m SSLError(e, request\u001b[39m=\u001b[39mrequest)\n\u001b[0;32m--> 565\u001b[0m     \u001b[39mraise\u001b[39;00m \u001b[39mConnectionError\u001b[39;00m(e, request\u001b[39m=\u001b[39mrequest)\n\u001b[1;32m    567\u001b[0m \u001b[39mexcept\u001b[39;00m ClosedPoolError \u001b[39mas\u001b[39;00m e:\n\u001b[1;32m    568\u001b[0m     \u001b[39mraise\u001b[39;00m \u001b[39mConnectionError\u001b[39;00m(e, request\u001b[39m=\u001b[39mrequest)\n",
      "\u001b[0;31mConnectionError\u001b[0m: HTTPSConnectionPool(host='www.soaringspot.com', port=443): Max retries exceeded with url: /en_gb/junior-world-gliding-championships-2022-tabor-2022/results (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x1054b8220>: Failed to establish a new connection: [Errno 8] nodename nor servname provided, or not known'))"
     ]
    }
   ],
   "source": [
    "import soaring_spot_dl as ss\n",
    "\n",
    "competition_url = \"/en_gb/junior-world-gliding-championships-2022-tabor-2022\"\n",
    "\n",
    "jwgc22 = ss.SoaringSpotDL(competition_url)\n",
    "jwgc22.download_traces()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "H8\n",
      "Start time: 12:58:22\n",
      "PEV: 12:49:37 UTC+2 49.28914N 14.50142E Press:5384ft GPS:5889ft\n",
      "------------------------------------\n",
      "SE\n",
      "Start time: 13:21:18\n",
      "PEV: 13:15:36 UTC+2 49.27278N 14.44111E Press:4859ft GPS:5111ft\n",
      "------------------------------------\n",
      "S1\n",
      "Start time: 13:09:18\n",
      "PEV: 13:00:45 UTC+2 49.27824N 14.36688E Press:5577ft GPS:5755ft\n",
      "------------------------------------\n",
      "JS\n",
      "Start time: 12:57:26\n",
      "PEV: 12:51:38 UTC+2 49.31167N 14.46808E Press:5751ft GPS:6237ft\n",
      "------------------------------------\n",
      "SB\n",
      "Start time: 13:09:09\n",
      "PEV: 13:00:42 UTC+2 49.27992N 14.35902E Press:5505ft GPS:5915ft\n",
      "------------------------------------\n",
      "JW\n",
      "Start time: 13:16:18\n",
      "PEV: 12:55:26 UTC+2 49.28133N 14.42786E Press:5804ft GPS:6286ft\n",
      "PEV: 13:07:08 UTC+2 49.27936N 14.43163E Press:5837ft GPS:6339ft\n",
      "------------------------------------\n",
      "KE\n",
      "Start time: 13:12:01\n",
      "PEV: 12:53:21 UTC+2 49.28608N 14.46779E Press:5777ft GPS:6302ft\n",
      "PEV: 13:06:24 UTC+2 49.27166N 14.45304E Press:5430ft GPS:5987ft\n",
      "------------------------------------\n",
      "DI\n",
      "Start time: 12:57:10\n",
      "PEV: 12:46:38 UTC+2 49.28918N 14.47074E Press:5361ft GPS:5899ft\n",
      "------------------------------------\n",
      "AU\n",
      "Start time: 13:03:05\n",
      "PEV: 12:51:39 UTC+2 49.28860N 14.50159E Press:5840ft GPS:6358ft\n",
      "------------------------------------\n",
      "Y\n",
      "Start time: 13:12:36\n",
      "PEV: 13:02:43 UTC+2 49.28189N 14.40721E Press:5705ft GPS:6204ft\n",
      "------------------------------------\n",
      "OK\n",
      "Start time: 13:09:15\n",
      "PEV: 13:00:52 UTC+2 49.27895N 14.36560E Press:5423ft GPS:5935ft\n",
      "------------------------------------\n",
      "71\n",
      "Start time: 13:08:34\n",
      "PEV: 12:48:54 UTC+2 49.28704N 14.47314E Press:5791ft GPS:6319ft\n",
      "PEV: 12:57:06 UTC+2 49.28334N 14.43365E Press:5823ft GPS:6345ft\n",
      "------------------------------------\n",
      "I\n",
      "Start time: 13:12:35\n",
      "PEV: 13:02:40 UTC+2 49.28164N 14.40659E Press:5712ft GPS:6204ft\n",
      "------------------------------------\n",
      "UB\n",
      "Start time: 13:14:18\n",
      "PEV: 13:03:35 UTC+2 49.28617N 14.47124E Press:5656ft GPS:6142ft\n",
      "------------------------------------\n",
      "HAL\n",
      "Start time: 12:59:34\n",
      "PEV: 12:51:25 UTC+2 49.29034N 14.48251E Press:5495ft GPS:5751ft\n",
      "------------------------------------\n",
      "X15\n",
      "Start time: 13:18:42\n",
      "PEV: 12:58:03 UTC+2 49.28628N 14.47054E Press:5640ft GPS:6135ft\n",
      "PEV: 13:11:36 UTC+2 49.28407N 14.46196E Press:5003ft GPS:5581ft\n",
      "------------------------------------\n",
      "MK\n",
      "Start time: 13:21:27\n",
      "PEV: 13:15:11 UTC+2 49.28473N 14.47859E Press:5656ft GPS:6135ft\n",
      "------------------------------------\n",
      "5F\n",
      "Start time: 13:29:29\n",
      "PEV: 13:22:20 UTC+2 49.27411N 14.43618E Press:5384ft GPS:5925ft\n",
      "------------------------------------\n",
      "44\n",
      "Start time: 13:06:20\n",
      "PEV: 12:48:57 UTC+2 49.27940N 14.42586E Press:5646ft GPS:6099ft\n",
      "PEV: 12:57:11 UTC+2 49.28200N 14.43088E Press:5830ft GPS:6260ft\n",
      "------------------------------------\n",
      "Z1\n",
      "Start time: 13:14:30\n",
      "PEV: 13:03:30 UTC+2 49.28473N 14.46871E Press:5607ft GPS:6155ft\n",
      "------------------------------------\n",
      "235\n",
      "Start time: 13:10:44\n",
      "PEV: 13:00:20 UTC+2 49.28364N 14.42527E Press:5607ft GPS:6132ft\n",
      "------------------------------------\n",
      "95\n",
      "Start time: 13:10:36\n",
      "PEV: 13:00:23 UTC+2 49.28165N 14.42750E Press:5771ft GPS:6250ft\n",
      "------------------------------------\n",
      "1X\n",
      "Start time: 13:16:26\n",
      "PEV: 12:55:22 UTC+2 49.27995N 14.41665E Press:5545ft GPS:6030ft\n",
      "PEV: 13:07:06 UTC+2 49.28165N 14.43003E Press:5863ft GPS:6348ft\n",
      "------------------------------------\n",
      "D2\n",
      "Start time: 13:05:34\n",
      "PEV: 12:49:12 UTC+2 49.27936N 14.42546E Press:5377ft GPS:5902ft\n",
      "PEV: 12:57:09 UTC+2 49.28383N 14.43200E Press:5604ft GPS:6135ft\n",
      "------------------------------------\n",
      "V92\n",
      "Start time: 13:18:22\n",
      "PEV: 12:58:02 UTC+2 49.28663N 14.47307E Press:5948ft GPS:6486ft\n",
      "PEV: 13:11:26 UTC+2 49.28175N 14.47010E Press:5302ft GPS:5837ft\n",
      "------------------------------------\n",
      "EU\n",
      "Start time: 13:02:37\n",
      "PEV: 12:51:40 UTC+2 49.28814N 14.50157E Press:5928ft GPS:6483ft\n",
      "------------------------------------\n",
      "2S\n",
      "Start time: 13:14:29\n",
      "PEV: 13:03:28 UTC+2 49.28566N 14.46892E Press:5761ft GPS:6263ft\n",
      "------------------------------------\n",
      "21\n",
      "Start time: 13:18:34\n",
      "PEV: 12:58:07 UTC+2 49.28689N 14.47298E Press:5705ft GPS:6197ft\n",
      "PEV: 13:11:31 UTC+2 49.28196N 14.46976E Press:5000ft GPS:5479ft\n",
      "------------------------------------\n",
      "SI\n",
      "Start time: 12:57:00\n",
      "PEV: 12:46:48 UTC+2 49.28778N 14.47184E Press:5535ft GPS:6050ft\n",
      "------------------------------------\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "{'H8': [<pev_data.Position at 0x1081a1820>],\n",
       " 'SE': [<pev_data.Position at 0x10f461fd0>],\n",
       " 'S1': [<pev_data.Position at 0x107e555e0>],\n",
       " 'JS': [<pev_data.Position at 0x10f488040>],\n",
       " 'SB': [<pev_data.Position at 0x10f488340>],\n",
       " 'JW': [<pev_data.Position at 0x10f488190>,\n",
       "  <pev_data.Position at 0x10f488520>],\n",
       " 'KE': [<pev_data.Position at 0x10f488550>,\n",
       "  <pev_data.Position at 0x10f4884c0>],\n",
       " 'DI': [<pev_data.Position at 0x10f488670>],\n",
       " 'AU': [<pev_data.Position at 0x10f488700>],\n",
       " 'Y': [<pev_data.Position at 0x10f488790>],\n",
       " 'OK': [<pev_data.Position at 0x10f488820>],\n",
       " '71': [<pev_data.Position at 0x10f4888b0>,\n",
       "  <pev_data.Position at 0x10f488880>],\n",
       " 'I': [<pev_data.Position at 0x10f4889d0>],\n",
       " 'UB': [<pev_data.Position at 0x10f488a60>],\n",
       " 'HAL': [<pev_data.Position at 0x10f488af0>],\n",
       " 'X15': [<pev_data.Position at 0x10f488b80>,\n",
       "  <pev_data.Position at 0x10f488b50>],\n",
       " 'MK': [<pev_data.Position at 0x10f488ca0>],\n",
       " '5F': [<pev_data.Position at 0x10f488d30>],\n",
       " '44': [<pev_data.Position at 0x10f488dc0>,\n",
       "  <pev_data.Position at 0x10f488d90>],\n",
       " 'Z1': [<pev_data.Position at 0x10f488ee0>],\n",
       " '235': [<pev_data.Position at 0x10f488f70>],\n",
       " '95': [<pev_data.Position at 0x10f488be0>],\n",
       " '1X': [<pev_data.Position at 0x10f488e20>,\n",
       "  <pev_data.Position at 0x10f4a9070>],\n",
       " 'D2': [<pev_data.Position at 0x10f4a9220>,\n",
       "  <pev_data.Position at 0x10f4a91f0>],\n",
       " 'V92': [<pev_data.Position at 0x10f4a9310>,\n",
       "  <pev_data.Position at 0x10f4a92e0>],\n",
       " 'EU': [<pev_data.Position at 0x10f4a9430>],\n",
       " '2S': [<pev_data.Position at 0x10f4a94c0>],\n",
       " '21': [<pev_data.Position at 0x10f4a9550>,\n",
       "  <pev_data.Position at 0x10f4a9520>],\n",
       " 'SI': [<pev_data.Position at 0x10f4a9670>]}"
      ]
     },
     "execution_count": 1,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import pev_data as pev\n",
    "\n",
    "pev.get_pevs(\"igc/Class STANDARD/2022-08-02\")\n",
    "\n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3.9.10 64-bit",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.13"
  },
  "orig_nbformat": 4,
  "vscode": {
   "interpreter": {
    "hash": "b0fa6594d8f4cbf19f97940f81e996739fb7646882a419484c72d19e05852a7e"
   }
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
